module/VuFind/src/VuFind/Controller/HoldsTrait.php

```diff
@@ -119,14 +123,17 @@ public function holdAction()
             // group, so make sure pickup locations match with the group
             $pickupDetails['requestGroupId'] = $requestGroups[0]['id'];
         }
-        $pickup = $catalog->getPickUpLocations($patron, $pickupDetails);
 
         // Check that there are pick up locations to choose from if the field is
         // required:
-        if (in_array('pickUpLocation', $extraHoldFields) && !$pickup) {
-            $this->flashMessenger()
-                ->addErrorMessage('No pickup locations available');
-            return $this->redirectToRecord('#top');
+        $pickup = [];
+        if (in_array('pickUpLocation', $extraHoldFields)) {
+            $pickup = $catalog->getPickUpLocations($patron, $pickupDetails);
+            if (!$pickup) {
+                $this->flashMessenger()
+                    ->addErrorMessage('No pickup locations available');
```

Comment:
While we're shaving off lines, I think it's safe to unwrap this one now that we have a longer line limit.

```suggestion
                $this->flashMessenger()->addErrorMessage('No pickup locations available');
```